bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableClusterUtilities.java

```diff
@@ -118,32 +184,73 @@ public synchronized ListClustersResponse getClusters() {
    * @param clusterId
    * @param zoneId
    * @param newSize
-   * @throws InterruptedException
+   * @throws InterruptedException if the cluster is in the middle of updating, and an interrupt was
+   *           received
+   */
+  public void setClusterSize(String clusterId, String zoneId, int newSize)
+      throws InterruptedException {
+    setClusterSize(getCluster(clusterId, zoneId), newSize);
+  }
+
+  /**
+   * Sets a cluster size to a specific size in an instance with a single clustr
+   * @param newSize
+   * @throws InterruptedException if the cluster is in the middle of updating, and an interrupt was
+   *           received
    */
-  public synchronized void setClusterSize(String clusterId, String zoneId, int newSize)
+  public void setClusterSize(int newSize) throws InterruptedException {
+    setClusterSize(getSingleCluster(), newSize);
+  }
+
+  private void setClusterSize(Cluster cluster, int newSize)
       throws InterruptedException {
     Preconditions.checkArgument(newSize > 0, "Cluster size must be > 0");
-    Cluster cluster = getCluster(clusterId, zoneId);
     int currentSize = cluster.getServeNodes();
     if (currentSize == newSize) {
-      logger.info("Cluster %s already has %d nodes.", clusterId, newSize);
+      logger.info("Cluster %s already has %d nodes.", getClusterId(cluster), newSize);
     } else {
       updateClusterSize(cluster.getName(), newSize);
     }
   }
 
+  public Cluster getSingleCluster() {
+    Preconditions.checkState(getClusters().getClustersCount() != 0,
+        "There must be at least one matching cluster for this method to work");
+    Preconditions.checkState(getClusters().getClustersCount() == 1,
+      "There can only be one cluster for this method to work.");
+    return getClusters().getClusters(0);
+  }
+
   /**
    * @param clusterId
    * @param zoneId
    * @param incrementCount a positive or negative number to add to the current node count
    * @return the new size of the cluster.
-   * @throws InterruptedException
+   * @throws InterruptedException if the cluster is in the middle of updating, and an interrupt was
+   *           received
    */
-  public synchronized int incrementClusterSize(String clusterId, String zoneId, int incrementCount)
+  public int incrementClusterSize(String clusterId, String zoneId, int incrementCount)
```

Comment:
I am not sure why we even have this method. Wouldn't setClusterSize() is sufficient.